agent/consul/state/memdb.go

```diff
@@ -140,7 +137,11 @@ type txn struct {
 	// Index is stored so that it may be passed along to any subscribers as part
 	// of a change event.
 	Index   uint64
-	publish func(tx ReadTxn, changes Changes) error
+	publish publishFuncType
+
+	prePublish prePublishFuncType
```

Comment:
Do we still need this to be a struct member, rather than just calling `processChanges` directly in `Commit`?

Reply:
Not sure that's feasible without major changes to the data struct, maybe I'm not understanding 🤔. Also, it could be handy to have the prePublish mockeable for future tests.